sklearn/cluster/tests/test_optics.py

```diff
@@ -784,7 +784,9 @@ def test_extract_dbscan():
 
 
 def test_precomputed_dists():
-    redX = X[::2]
+    # This slicing makes the array F-ordered.
+    # but we need C-ordering.
+    redX = np.ascontiguousarray(X[::2])
```

Comment:
This call to `np.ascontiguousarray` should happen transparently inside the `pairwise_distances` implementation (and only for the dense case).

Reply:
Would it be better that this conversion happens in `PairwiseDistancesReduction.__init__` [as suggested here](https://github.com/scikit-learn/scikit-learn/pull/20254/files#r670616290) ?

Reply:
Reverted by 4c237515ba3d0232f077d0dbb03028c21eb36d15.